### DISCUSSION TITLE
@types/jasmine - tsc errors

### DISCUSSION THREAD

**Discussion by rebajz** [DefinitelyTyped/DefinitelyTyped#56774] [Issues with a @types package]
Hi,
i have:
"typescript": "4.1.6",
"jasmine": "3.10.0",
"@types/jasmine": "3.4.0",
nodejs 14.18.0 32bit

this working correctly...

When i upgrade @types/jasmine to another up
for example "@types/jasmine": "3.9.1",

i got this errors after tsc:
> tsc

node_modules/@types/jasmine/index.d.ts:831:15 - error TS2430: Interface 'FunctionMatchers<Fn>' incorrectly extends interface 'Matchers<any>'.
  The types returned by 'toHaveBeenCalledWith(...)' are incompatible between these types.
    Type 'void' is not assignable to type 'Promise<void>'.

831     interface FunctionMatchers<Fn extends Func> extends Matchers<any> {
                  ~~~~~~~~~~~~~~~~

Workflows/CEF.ts:340:26 - error TS2339: Property 'describe' does not exist on type 'Env'.

340         jasmine.getEnv().describe = (() => {
                             ~~~~~~~~

Workflows/CEF.ts:341:41 - error TS2339: Property 'describe' does not exist on type 'Env'.

341             let func = jasmine.getEnv().describe;
                                            ~~~~~~~~


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 2

**Comment by samarmohan** [comment]
Looking at [`types/jasmine/index.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jasmine/index.d.ts), I don't see a problem in the code. Have you tried deleting `node_modules` and re installing?

**Comment by rebajz** [comment]
yes, of course i tried it before i wrote my post.

and here is code started on line 340

        jasmine.getEnv().describe = (() => {
            let func = jasmine.getEnv().describe;
            return function () {
                let res = func.apply(this, arguments);
                suitesArr.push(res);
                return res;
            };
        })();
        setInterval(fatalErrorHandling, fatalErrorPollingInterval);

[a comment by an automated account is not shown]
